src/GafferScene/Set.cpp

```diff
@@ -295,42 +295,43 @@ IECore::ConstInternedStringVectorDataPtr Set::computeSetNames( const Gaffer::Con
 		return inNamesData;
 	}
 
-	vector<InternedString> tokenizedNames;
-	StringAlgo::tokenize( names, ' ', tokenizedNames );
-
-	// specific logic if we have only one item, to avoid the more complex logic of adding two lists together
-	if( tokenizedNames.size() == 1 ) {
-		const std::vector<InternedString> &inNames = inNamesData->readable();
-		if( std::find( inNames.begin(), inNames.end(), tokenizedNames[0] ) != inNames.end() )
+	// Fast path for the common case of a single name.
+	const std::vector<InternedString> &inNames = inNamesData->readable();
+	if( names.find( ' ' ) == string::npos )
+	{
+		if( StringAlgo::hasWildcards( names ) || find( inNames.begin(), inNames.end(), names ) != inNames.end() )
 		{
 			return inNamesData;
 		}
-
 		InternedStringVectorDataPtr resultData = inNamesData->copy();
-		resultData->writable().push_back( tokenizedNames[0] );
+		resultData->writable().push_back( names );
 		return resultData;
 	}
 
-	// inserting the new names into the vector
```

Comment:
As discussed this morning:  ignoring wildcards makes sense in Add or Remove, but doesn't really make any sense for Create mode.  Perhaps this should be an error?  We decided to split this one commit out of this PR so it can be deferred to Gaffer 0.60 so we have more time to think about it, which is kind of a good idea anyway since this is a compatibility break if anyone is silly enough to have "*" in their set name.